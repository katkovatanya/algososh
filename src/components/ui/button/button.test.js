import renderer from 'react-test-renderer';
import { Button } from './button';
import { render, screen, fireEvent } from "@testing-library/react";

describe('Тестирование компонента Button', function () {
  it('кнопка с текстом', function () {
    const element = renderer.create(<Button text="тест" />).toJSON();
    expect(element).toMatchSnapshot();
  });
  it('кнопка без текста', function () {
    const element = renderer.create(<Button text="" />).toJSON();
    expect(element).toMatchSnapshot();
  });
  it('заблокированная кнопка', function () {
    const element = renderer.create(<Button text="тест" disabled={true} />).toJSON();
    expect(element).toMatchSnapshot();
  });
  it('кнопка без текста', function () {
    const element = renderer.create(<Button text="тест" isLoader={true} />).toJSON();
    expect(element).toMatchSnapshot();
  });
  it('Проверка колбека', function () {
    const testCallback = jest.fn();
    render(<Button text="TEST" onClick={testCallback} />);
    fireEvent.click(screen.getByText('TEST'));
    expect(testCallback).toHaveBeenCalled();
  });
})