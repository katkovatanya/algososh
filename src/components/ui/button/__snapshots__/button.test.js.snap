// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Button заблокированная кнопка 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    тест
  </p>
</button>
`;

exports[`Тестирование компонента Button кнопка без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    
  </p>
</button>
`;

exports[`Тестирование компонента Button кнопка без текста 2`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование компонента Button кнопка с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    тест
  </p>
</button>
`;
